Content.Shared/Xenoarchaeology/Artifact/XAE/XAEShuffleSystem.cs

```diff
@@ -0,0 +1,44 @@
+using Content.Shared.Mobs.Components;
+using Content.Shared.Xenoarchaeology.Artifact.XAE.Components;
+using Robust.Shared.Random;
+using Robust.Shared.Timing;
+
+namespace Content.Shared.Xenoarchaeology.Artifact.XAE;
+
+public sealed class XAEShuffleSystem : BaseXAESystem<XAEShuffleComponent>
+{
+    [Dependency] private readonly EntityLookupSystem _lookup = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly SharedTransformSystem _xform = default!;
+    [Dependency] private readonly IGameTiming _timing = default!;
+
+    /// <inheritdoc />
+    protected override void OnActivated(Entity<XAEShuffleComponent> ent, ref XenoArtifactNodeActivatedEvent args)
+    {
+        if(!_timing.IsFirstTimePredicted)
+            return;
+
+        var mobState = GetEntityQuery<MobStateComponent>();
+
+        List<Entity<TransformComponent>> toShuffle = new();
+
+        foreach (var entity in _lookup.GetEntitiesInRange(ent.Owner, ent.Comp.Radius, LookupFlags.Dynamic | LookupFlags.Sundries))
+        {
+            if (!mobState.HasComponent(entity))
+                continue;
+
+            var xform = Transform(entity);
+
+            toShuffle.Add((entity, xform));
+        }
+
+        _random.Shuffle(toShuffle);
+
+        while (toShuffle.Count > 1)
+        {
+            var ent1 = _random.PickAndTake(toShuffle);
```

Comment:
This will do less than you think it will; there's a reasonable chance that the shuffling will impact a previously-shuffled person, or indeed shuffle the same person with themselves. So it's unlikely it'll shuffle everyone.

This probably should pop the shuffle-ees off the list so they can't be moved more than once.

Reply:
i'm pretty sure thats just the way old system is working... can leave todo and should add it to the issue of this refactor summary, i guess...

Reply:
I personally think this should be removed, deleted to the ground. We should have shuffle mind effect but not shuffle bodies..